pypeit/spectrographs/spectrograph.py

```diff
@@ -342,12 +382,15 @@ def empty_bpm(self, shape=None, filename=None, det=1):
               0=not masked; 1=masked
 
         """
+
+        # TODO The logic heere is botched. shape has to be set for the code not to crash so why is filename even
```

Comment:
because the evolution was not done whole-sale. 
I think we can remove it now.

Reply:
actually some instruments still use the filename like LRIS_red. Anyway, we should clean this up.